### PR TITLE
Enable running interop tests with dotnet CLI + CoreCLR

### LINKED ISSUE #8014: Enable running interop tests under CoreCLR
-- adjust run_interop_tests.py to allow this.
-- how to keep both .NET and coreclr tests? there's only one "csharp" language.

### PR DESCRIPTION
**Pull request by jtattermusch**
Based on #8039 (to be merged first)

Fixes #8014.

### CODE REVIEW

File: tools/dockerfile/interoptest/grpc_interop_csharpcoreclr/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+RUN echo "deb http://download.mono-project.com/repo/debian wheezy-libtiff-compat main" | tee -a /etc/apt/sources.list.d/mono-xamarin.list
+
+# Install dependencies
+RUN apt-get update && apt-get -y dist-upgrade && apt-get install -y \
+    mono-devel \
+    ca-certificates-mono \
+    nuget \
+    && apt-get clean
+
+RUN nuget update -self
```
Comment (apolcyn) [inline tools/dockerfile/interoptest/grpc_interop_csharpcoreclr/Dockerfile]: Probably here only because of the unchanged `charp_deps.include` file. Somewhat unrelated to change here though, could probably be changed in a separate PR.

File: src/csharp/Grpc.IntegrationTesting.Client/project.json [hunk 2]
```diff
@@ -13,8 +13,10 @@
           "includeFiles": [ "../Grpc.Core/Version.cs" ]
         },
         "copyToOutput": {
-          "include": "data/*",
```
Comment (apolcyn) [inline src/csharp/Grpc.IntegrationTesting.Client/project.json]: I think it makes sense how this would not work with dotnet cli, as it looks to bet trying to copy the data/ directory from the separate project, Grpc.IntegrationTesting, but it seems strange how this works with xbuild.

File: templates/tools/dockerfile/csharp_dotnetcli_deps.include [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Install dotnet SDK based on https://www.microsoft.com/net/core#debian
+RUN apt-get update && apt-get install -y curl libunwind8 gettext
+RUN curl -sSL -o dotnet.tar.gz https://go.microsoft.com/fwlink/?LinkID=809130
+RUN mkdir -p /opt/dotnet && tar zxf dotnet.tar.gz -C /opt/dotnet
+RUN ln -s /opt/dotnet/dotnet /usr/local/bin
+
+# Trigger the population of the local package cache
+ENV NUGET_XMLDOC_MODE skip
+RUN mkdir warmup ${'\\'}
```
Comment (apolcyn) [inline templates/tools/dockerfile/csharp_dotnetcli_deps.include]: correct me if im wrong, i think this is only to download common package now for quicker runs later

**Comment by apolcyn** [comment]
LGTM.

I don't think the test failures here are related, but it looks like the new coreclr interop tests are passing.

File: tools/dockerfile/interoptest/grpc_interop_csharpcoreclr/Dockerfile (hunk 1 above)
Comment (jtattermusch) [inline tools/dockerfile/interoptest/grpc_interop_csharpcoreclr/Dockerfile]: yes, let's get rid of nuget update -self   in  run_tests.py and run_interop_tests.py separately.

File: templates/tools/dockerfile/csharp_dotnetcli_deps.include (hunk 3 above)
Comment (jtattermusch) [inline templates/tools/dockerfile/csharp_dotnetcli_deps.include]: Yes, you are right. Otherwise dotnet restore would go ahead and download a good portion of the framework libraries (with old .NET it would just download a few packages). So I think this saves the number of downloads significantly.
